### ISSUE TITLE
Use OpenStreetMap as location provider

### ISSUE THREAD

**Issue by piratenpanda** [WhisperSystems/Signal-Android#5042]
I think in most areas of the world OpenStreetMap provides better and more detailled map data than Google Maps. Also there are many providers you can get map data and have an osm map underneath if osm.org or nominatim alone don't work for how location sharing is implemented today. i
I would love to see more love for non proprietary stuff.

**Comment by FeuRenard** [comment]
Something on this topic was already said in #4980.

**Comment by moxie0** [comment]
Thanks but I don't think we're going to do this.  We'd have to rewrite the entire placepicker, and it looks like the OSM Android SDK is a ways behind in MapView compatibility.

**Comment by amenk** [comment]
So sad :-(

**Comment by Be-ing** [comment]
I have opened a [feature request](https://github.com/osmandapp/Osmand/issues/3526) for OsmAnd to provide a drop in replacement for the Google Maps API.

FWIW, I can receive locations other Signal users send to me and open them with OsmAnd without Google Maps.

**Comment by Be-ing** [comment]
OsmAnd has an [API](http://osmand.net/build_it) to hook into. I do not know whether it provides everything that would be needed for Signal's use case. Unfortunately using it would require writing a new fallback place picker from scratch.

**Comment by Be-ing** [comment]
@moxie0 would you accept a pull request that used OsmAnd's API as a fallback if Google Maps is not available?

**Comment by Killercat103** [comment]
Ive noticed that the mobile Element Matrix client utilizes Maplibre for location shating. I found this interface they use quite pleasant and intuitive to use.

Perhaps Element could be an inspiration? I figured this sdk could be of use to Signal.

 https://maplibre.org/projects/maplibre-native/
 
I am not sure if feature bounties are a thing to offer but I would place a small one for OSM location sharing and F-Droid support if I knew how.
